store/badger.go

```diff
@@ -41,3 +44,35 @@ func (b *BadgerKV) Delete(key []byte) error {
 	}
 	return txn.Commit()
 }
+
+func (b *BadgerKV) NewBatch() Batch {
+	//write transactions should be short lived as they use extra resources in badger
+	return &BadgerBatch{
+		txn: b.db.NewTransaction(true),
+	}
+}
+
+type BadgerBatch struct {
+	txn *badger.Txn
+}
+
+func (bb *BadgerBatch) Set(key, value []byte) error {
+	if err := bb.txn.Set(key, value); err != nil {
+		bb.Discard()
+		return err
+	}
```

Comment:
I think we should just propagate error, and let user decide, wether to `Discard` or not.